armadillo/src/main/java/at/favre/lib/armadillo/FixedBcryptKeyStretcher.java

```diff
@@ -58,24 +68,23 @@
      * @param logRounds log2(Iterations). e.g. 12 ==> 2^12 = 4,096 iterations
      * @return the Bcrypt hash of the password
      */
-    private static byte[] bcrypt(char[] password, byte[] salt, int logRounds) {
+    private static byte[] bcrypt(byte[] salt, char[] password, int logRounds) {
         StrictMode.noteSlowCall("bcrypt is a very expensive call and should not be done on the main thread");
-        return Bytes.from(BCrypt.hashpw(String.valueOf(password) + Bytes.wrap(salt).encodeHex(), generateSalt(salt, logRounds))).array();
+        byte[] passwordBytes = null;
+        try {
+            passwordBytes = charArrayToByteArray(password, StandardCharsets.UTF_8);
+            return BCrypt.withDefaults().hash(logRounds,
```

Comment:
I was thinking, maybe we just use the raw bcrypt hash her?
Background: Everybody implements bcrypt with the Modular Hash Format used by OpenBSD which is:

    ${version}${log-count}${16-byte-salt-radix64}{23-byte-hash-radix64}

e.g.

    $2y$06$doGnefu9cbLkJTn8sef7U.dynHJFe5hS6xp7vLWb2Zu7e8cOuMVmS

This makes sense for password hash for storage, but we provide the salt ourselves, so using the everything BUT the hash seems sloppy? Maybe there are some drawbacks I dont think about right now?

Reply:
I agree that using the full bcrypt message has no value in this situation, as you are not storing it anywhere. I don't see any drawbacks.